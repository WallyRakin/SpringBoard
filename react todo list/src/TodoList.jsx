// TodoList.js
import React, { useState } from 'react';
import { NewTodoForm } from './NewTodoForm';
import { Todo } from './Todo';

export function TodoList() {
    const [todos, setTodos] = useState([]);

    const addTodo = (newTodo) => {
        setTodos([...todos, newTodo]);
    };

    const removeTodo = (id) => {
        setTodos(todos.filter(todo => todo.id !== id));
    };

    return (
        <div>
            <NewTodoForm addTodo={addTodo} />
            {todos.map(todo => (
                <Todo key={todo.id} id={todo.id} task={todo.task} removeTodo={removeTodo} />
            ))}
        </div>
    );
};