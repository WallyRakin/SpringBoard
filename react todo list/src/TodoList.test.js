import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { TodoList } from './TodoList';

it('TodoList renders without crashing', () => {
    render(<TodoList />);
});

it('TodoList snapshot', () => {
    const tree = renderer.create(<TodoList />).toJSON();
    expect(tree).toMatchSnapshot();
});
