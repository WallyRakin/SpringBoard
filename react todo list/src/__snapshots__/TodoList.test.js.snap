// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Enter a new task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</div>
`;
